### PR TITLE
3P Media: Use videos instead of gifs in library

### LINKED ISSUE #5006: 3P Media: Use videos instead of gifs in library
## Feature Description

This might have a positive impact on performance as otherwise loading tons of large GIFs in the library is quite a heavy task.

See #5005, #4885

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by embarks**
## Context
- Try displaying the gifs as videos to enhance the performance of the Tenor gif media tab.

## Summary
- Change the elements which display the images in the 3P media tab to display the Tenor gifs as videos.

## Relevant Technical Choices
- `InnerElement` is the renderer for the images in the 3P media tab. I changed the render function there so the `gif` would display as a video (with both webm/mp4s from Tenor)
- The videos are in the `resource.output.sizes` property from past work to display gifs as videos in the published output. I reuse those here to display them as videos in the 3P media tab.
- The gif's poster is in the `output` property and comes from Tenor. I use that in the same way as the Coverr video-- instead of waiting for the video to load to show the poster, use the poster's `onLoad` to show the preview/poster. I edited `mediaElement` so the media would play when hovering. 
- The gif poster is also used as the ghost image when being dragged.

## To-do
- N/A

## User-facing changes
- The gif video resources now only play on hover

### QA

This is a performance enhancement and should have no impact on the user other than loading and showing the posters in the tab should be less heavy than before, so I did not include a QA step. There has been a visual change in that the gifs no longer automatically play, so that should be validated. 

- [x] This is a non-user-facing change and requires no QA

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #5006

### CODE REVIEW

**Review by maxyinger — APPROVED** [review]
looks great!

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 1]
```diff
@@ -104,88 +104,110 @@ function InnerElement({
...
-      />
-    );
+  const imageProps = {
+    ...cloneProps,
+    src: thumbnailURL,
+    onLoad: makeMediaVisible,
+  };
+  const videoProps = {
+    ...commonProps,
+    loop: type === 'gif',
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js]: smart

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 2]
```diff
@@ -104,88 +104,110 @@ function InnerElement({
...
+    poster: displayPoster,
+    showWithoutDelay: Boolean(newVideoPosterRef.current),
+  };
+
+  if (type === 'image') {
+    media = <Image key={src} {...imageProps} ref={mediaElement} />;
     cloneProps.src = thumbnailURL;
-  } else if (type === 'video') {
-    const displayPoster = poster ? poster : newVideoPosterRef.current;
+  } else if (['gif', 'video'].includes(type)) {
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js]: you reference `['gif', 'video']` a few times, any use in pulling this out to make it a constant?

**Review by BrittanyIRL — COMMENTED** [review]
It appears that the videos are only playing on hover or focus, is this intentional? 

only play on hover

**Comment by embarks** [comment]
> It appears that the videos are only playing on hover or focus, is this intentional?
> 
> only play on hover

@brittany there weren't any requirements here, I just copied the video. I think I can do it either way. I thought this looked less chaotic but it might not be expected.

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js (hunk 2 above)
Comment (embarks) [reply assets/src/edit-story/components/library/panes/media/common/innerElement.js]: Sure, the element type string literals might be in some more places but I can at least add them in the files I touched and I'll remember for future reviews to replace them or remind others to replace them.

**Comment by swissspidy** [comment]
> > It appears that the videos are only playing on hover or focus, is this intentional?
> > only play on hover
> 
> @brittany there weren't any requirements here, I just copied the video. I think I can do it either way. I thought this looked less chaotic but it might not be expected.

Good callout. Let's ask for clarification!

@o-fernandez @samitron7 As we are now using videos elements instead of GIF images for performance reasons, it is possible to only animate those GIFs on hover (see demo above). This is the same behavior as for videos in the local media library. Should this stay this way, or should these GIF videos autplay all the time?

I lean towards keeping existing behavior as today (always autoplay), but it does feel a bit chaotic.

Play-on-hover is slightly better for performance. Although either solution is already much more performant than what we have today.

**Comment by o-fernandez** [comment]
> > > It appears that the videos are only playing on hover or focus, is this intentional?
> > > only play on hover
> > 
> > 
> > @brittany there weren't any requirements here, I just copied the video. I think I can do it either way. I thought this looked less chaotic but it might not be expected.
> 
> Good callout. Let's ask for clarification!
> 
> @o-fernandez @samitron7 As we are now using videos elements instead of GIF images for performance reasons, it is possible to only animate those GIFs on hover (see demo above). This is the same behavior as for videos in the local media library. Should this stay this way, or should these GIF videos autplay all the time?
> 
> I lean towards keeping existing behavior as today (always autoplay), but it does feel a bit chaotic.
> 
> Play-on-hover is slightly better for performance. Although either solution is already much more performant than what we have today.

I'm OK with play on hover. We can monitor to see if we get any negative user feedback after the change, but I expect this to be an OK implementation.

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 3]
```diff
@@ -104,11 +105,14 @@ function InnerElement({
...
   useAverageColor(
-    type === 'video' ? hiddenPoster : mediaElement,
+    [ContentType.VIDEO, ContentType.GIF].includes(type)
+      ? hiddenPoster
+      : mediaElement,
     setAverageColor
   );
 
   useEffect(() => {
+    // assign display poster for videos
```
Comment (samwhale) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js:115]: 👍

**Review by samwhale — APPROVED** [review]
Nice, this is great ✨

**Review by BrittanyIRL — APPROVED** [review]

**Pull request merged**